src/main/java/com/github/junrar/rarfile/FileHeader.java

```diff
@@ -186,9 +192,71 @@ public FileHeader(BlockHeader bh, byte[] fileHeader) {
                 position++;
             }
         }
-        mTime = getDateDos(fileTime);
-        // TODO rartime -> extended
 
+        mTime = FileTime.fromMillis(getDateDos(fileTime));
+
+        if (hasExtTime()) {
+            short extTimeFlags = Raw.readShortLittleEndian(fileHeader, position);
+            position += 2;
+
+            TimePositionTuple mTimeTuple = parseExtTime(12, extTimeFlags, fileHeader, position, mTime);
+            mTime = mTimeTuple.time;
+            position = mTimeTuple.position;
+
+            TimePositionTuple cTimeTuple = parseExtTime(8, extTimeFlags, fileHeader, position);
+            cTime = cTimeTuple.time;
+            position = cTimeTuple.position;
+
+            TimePositionTuple aTimeTuple = parseExtTime(4, extTimeFlags, fileHeader, position);
+            aTime = aTimeTuple.time;
+            position = aTimeTuple.position;
+
+            TimePositionTuple arcTimeTuple = parseExtTime(0, extTimeFlags, fileHeader, position);
+            arcTime = arcTimeTuple.time;
+            position = arcTimeTuple.position;
+        }
+    }
+
+    private static final class TimePositionTuple {
+        private final int position;
+        private final FileTime time;
+
+        private TimePositionTuple(int position, FileTime time) {
+            this.position = position;
+            this.time = time;
+        }
+    }
+
+    private static TimePositionTuple parseExtTime(int shift, short flags, byte[] fileHeader, int position) {
+        return parseExtTime(shift, flags, fileHeader, position, null);
+    }
+
+    private static TimePositionTuple parseExtTime(int shift, short flags, byte[] fileHeader, int position, FileTime baseTime) {
+        int flag = flags >>> shift;
+        if ((flag & 0x8) != 0) {
+            long seconds;
+            if (baseTime != null) {
+                seconds = baseTime.toMillis() / MILLIS_PER_SECOND;
```

Comment:
I personally would prefer to use the Java TimeUnit class to do this conversion here and in the else below. Code would look like this then
```suggestion
                seconds = TimeUnit.MILLISECONDS.toSeconds(baseTime.toMillis());
```
I leave it up to you since this is only cosmetic

Reply:
I think FileTime has a `to(TimeUnit)` method that I could just have used for that. It slipped my mind.